tests/feature/test_outline.py

```diff
@@ -320,3 +320,70 @@ def _(site):
     result = pytester.runpytest("-s")
     result.assert_outcomes(passed=1)
     assert collect_dumped_objects(result) == ["https://my-site.com"]
+
+
+def test_variable_reuse(pytester):
+    """
+    Test example parameter name and step arg do not redefine each other's value
+    if the same name is used for both in different steps.
+    """
+
+    pytester.makefile(
+        ".feature",
+        outline=textwrap.dedent(
+            """\
+            Feature: Example parameters reuse
+                Scenario Outline: Check for example parameter re-use
+                    Given some <key> exists
+                    When I print <css_id>
+                    And I output "some value"
+                    And I echo <css_id>
+                    Then finish testing
+
+                    Examples:
+                        | key  | css_id |
+                        | foo  | bar    |
+                        | foo2 | bar2   |
```

Comment:
can we compress this test to the minimal amount of steps necessary?
Also, let's use more of `dump_obj` and `collect_dumped_objects` instead of asserting in `Then` steps.
This way if the `Then` functionality is broken and passes even though it shouldn't, we would still be doing the assertions.

Reply:
Good point - my bad for just copying the test from the issue as a starting point!